### DISCUSSION TITLE
Full Stack JavaScript feedback. Intermediate HTML and CSS Course

### DISCUSSION THREAD

**Discussion by JustDom223** [TheOdinProject/curriculum#26884] [Ideas]
### Describe your suggestion

I'm thoroughly enjoying my course journey, but as I progress through the Intermediate HTML and CSS section, I've encountered a noticeable gap in the learning experience. While I appreciate the valuable information provided in MDN webdocs and other materials, I've found the extensive reading requirements, especially before tackling the 'Form' task, a bit overwhelming.

Trying to retain all of these sections without any solid application was a doozy:

[Intermediate CSS Concepts](https://www.theodinproject.com/paths/full-stack-javascript/courses/intermediate-html-and-css#intermediate-css-concepts)
[Default Styles](https://www.theodinproject.com/lessons/node-path-intermediate-html-and-css-default-styles)

[CSS Units](https://www.theodinproject.com/lessons/node-path-intermediate-html-and-css-css-units)

[More Text Styles](https://www.theodinproject.com/lessons/node-path-intermediate-html-and-css-more-text-styles)

[More CSS Properties](https://www.theodinproject.com/lessons/node-path-intermediate-html-and-css-more-css-properties)

[Advanced Selectors](https://www.theodinproject.com/lessons/node-path-intermediate-html-and-css-advanced-selectors)

[Positioning](https://www.theodinproject.com/lessons/node-path-intermediate-html-and-css-positioning)

[CSS Functions](https://www.theodinproject.com/lessons/node-path-intermediate-html-and-css-css-functions)

[Custom Properties](https://www.theodinproject.com/lessons/node-path-intermediate-html-and-css-custom-properties)

[Browser Compatibility](https://www.theodinproject.com/lessons/node-path-intermediate-html-and-css-browser-compatibility)

[Frameworks and Preprocessors](https://www.theodinproject.com/lessons/node-path-intermediate-html-and-css-frameworks-and-preprocessors)

[Forms](https://www.theodinproject.com/paths/full-stack-javascript/courses/intermediate-html-and-css#forms)
[Form Basics](https://www.theodinproject.com/lessons/node-path-intermediate-html-and-css-form-basics)

[Form Validation](https://www.theodinproject.com/lessons/node-path-intermediate-html-and-css-form-validation)

[Project: Sign-up Form](https://www.theodinproject.com/lessons/node-path-intermediate-html-and-css-sign-up-form)

I believe that breaking down this section with smaller, practical tasks that align with the theoretical content would be beneficial. These tasks could serve as opportunities to reinforce the concepts learned. The pacing and structure of the earlier foundational sections seemed more effective in aiding my understanding and retention.

I genuinely value the course and my overall progress, though it has been at different rates for various sections. 

### Path

Node / JS

### Lesson Url

https://www.theodinproject.com/paths/full-stack-javascript/courses/intermediate-html-and-css

### Checks

- [X] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
- [ ] Would you like to work on this issue?

### (Optional) Additional Comments

I have looked over the existing issues and didn't see any mention or clashing issues. Please let me know if this has been raised correctly.

**Comment by CouchofTomato** [comment]
@TheOdinProject/javascript 

Can we get an opinion on this. Might be better to move to a discussion for now.

**Comment by radtradcath** [comment]
I don't think you're supposed to memorize everything that was taught in these lessons at first glance.

Every information you need to advance in this section is presented in one of the lessons, so you can just go back to thr once you need to remember something.

OP compared Interm. HTML & CSS with Foundations. But it's natural that content will get denser and harder, since now we're growing in complexity.

Memorization will come from usage of the features your learned about. Also, assignments from this section are amazingly good. I'd say it's fine the way it is. I remember from my time spent in this very section that this was when I started really progressing on how to do my own research for questions I had.

**Comment by Anacodes123** [comment]
Creating simple projects on my own helps me a lot in understanding how HTML and CSS works. In this way, my learning progress is faster compared to just watching YT tutorials.

**Comment by JustDom223** [comment]
> I don't think you're supposed to memorize everything that was taught in these lessons at first glance.
> 
> Every information you need to advance in this section is presented in one of the lessons, so you can just go back to thr once you need to remember something.
> 
> OP compared Interm. HTML & CSS with Foundations. But it's natural that content will get denser and harder, since now we're growing in complexity.
> 
> Memorization will come from usage of the features your learned about. Also, assignments from this section are amazingly good. I'd say it's fine the way it is. I remember from my time spent in this very section that this was when I started really progressing on how to do my own research for questions I had.

While I do understand where you are coming from. 
I am now onto the 'JavaScript course' and it has been broken up into 4 projects each with 2-4 lessons before to help reinforce the different things that you are learning. This is exactly how almost all of the other sections are set up. 
The fact that there are 16 different lessons before the first project is my reason for finding this section lacking in reinforcement.

When I have more time I will try to shed more light on a way that I think it could be improved on.

**Comment by radtradcath** [comment]
> > I don't think you're supposed to memorize everything that was taught in these lessons at first glance.
> > 
> > Every information you need to advance in this section is presented in one of the lessons, so you can just go back to thr once you need to remember something.
> > 
> > OP compared Interm. HTML & CSS with Foundations. But it's natural that content will get denser and harder, since now we're growing in complexity.
> > 
> > Memorization will come from usage of the features your learned about. Also, assignments from this section are amazingly good. I'd say it's fine the way it is. I remember from my time spent in this very section that this was when I started really progressing on how to do my own research for questions I had.
> 
> While I do understand where you are coming from. 
> I am now onto the 'JavaScript course' and it has been broken up into 4 projects each with 2-4 lessons before to help reinforce the different things that you are learning. This is exactly how almost all of the other sections are set up. 
> The fact that there are 16 different lessons before the first project is my reason for finding this section lacking in reinforcement.
> 
> When I have more time I will try to shed more light on a way that I think it could be improved on. 

I'm sure a sctructured suggestion for improvement would help clearing this matter. There's a high chance that I'm missing something, since it's been a while since I completed this section.
